index.js

```diff
@@ -62,200 +51,117 @@ const Prompts = (function () {
   };
 
   function applyStyles(element, styles) {
-    for (const prop in styles) {
-      element.style[prop] = styles[prop];
-    }
+    Object.assign(element.style, styles);
   }
 
-  function createModal(message) {
-    const overlay = document.createElement("div");
-    applyStyles(overlay, overlayStyle);
-
-    const modal = document.createElement("div");
-    applyStyles(modal, modalStyle);
+  function createDialog(message) {
+    const dialog = document.createElement("dialog");
+    applyStyles(dialog, dialogStyle);
+    dialog.setAttribute("role", "dialog");
+    dialog.setAttribute("aria-modal", "true");
 
     const form = document.createElement("form");
-    form.setAttribute("novalidate", "true");
-    modal.appendChild(form);
+    form.method = "dialog"; // Allows form to close the dialog on submission.
```

Comment:
I hadn't seen this before: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#method

> `method="dialog"`: When the form is inside a [`<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog), closes the dialog and causes a `submit` event to be fired on submission, without submitting data or clearing the form.